src/components/RadioButton/RadioButton.tsx

```diff
@@ -1,73 +1,65 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import classnames from 'classnames/bind';
 import styles from './RadioButton.scss';
 
 const cx = classnames.bind( styles );
 
-const RadioButton = ( {
-	checked,
-	children,
-	disabled,
-	display,
-	name,
-	onChange,
-	required,
-} ) => (
-	<label className={cx( 'container', display )}>
-		<input
-			checked={checked}
-			className={cx( 'input', display )}
-			disabled={disabled}
-			name={name}
-			onChange={onChange}
-			required={required}
-			type="radio"
-		/>
-		{children}
-	</label>
-);
-
-RadioButton.propTypes = {
+function RadioButton ( props: {
 	/**
 	 * Toggle for the component’s checked state. Forwarded to the input
 	 * element.
 	 */
-	checked: PropTypes.bool,
+	checked: boolean,
 	/**
 	 * Child nodes to be rendered as the label for the radio button.
 	 * Label/input HTML linking happens automatically.
 	 */
-	children: PropTypes.node,
+	children: React.ReactNode,
 	/**
 	 * Whether the radio button is interactive. Forwarded to the input
 	 * element.
 	 */
-	disabled: PropTypes.bool.isRequired,
+	disabled: boolean,
 	/**
 	 * Controls the DOM flow behavior of the component.
 	 */
-	display: PropTypes.oneOf( [ 'block', 'inline' ] ),
+	display: 'block' | 'inline',
 	/**
 	 * A string to combine multiple radio buttons in order to provide a
 	 * multiple choice field. Forwarded to the input element.
 	 */
-	name: PropTypes.oneOfType( [
-		PropTypes.number,
-		PropTypes.string,
-	] ),
+	name: string,
 	/**
 	 * Function to run when the radio button is toggled. Forwarded to the
 	 * input element.
 	 */
-	onChange: PropTypes.func,
+	onChange: ( event: React.ChangeEvent ) => void,
+	/**
+      * Forward a ref to the element.
+      */
+	ref?: React.Ref<HTMLInputElement>
 	/**
 	 * Indicates that the radio button must be checked by the user in
 	 * order for the parent form to validate. Forwarded to the input
 	 * element.
 	 */
-	required: PropTypes.bool.isRequired,
-};
+	required: boolean,
+} ) {
+	return (
+		<label className={cx( 'container', props.display )}>
+			<input
+				checked={props.checked}
+				className={cx( 'input', props.display )}
+				disabled={props.disabled}
+				name={props.name}
+				onChange={props.onChange}
+				required={props.required}
+				type="radio"
+			/>
+			{props.children}
+		</label>
+	);
+}
 
 RadioButton.defaultProps = {
```

Comment:
these props are all required, so the defaultProps serve no purpose

Reply:
if you wanted to make them optional you could, but instead of `defaultProps`, destructure from props inside the function and assign default values, e.g.:

```
const { disabled = false } = props;
```

Reply:
Ah interesting - I believed that they would assign defaults even in the event of an unpassed value, but I think I inferred that wrongly. Will update!